google/cloud/bigquery/__init__.py

```diff
@@ -126,12 +126,12 @@
 
 if sys_major == 3 and sys_minor in (7, 8):
     warnings.warn(
-        "The python-bigquery library will stop supporting Python 3.7 "
-        "and Python 3.8 in a future major release expected in Q4 2024. "
+        "The python-bigquery library no longer supports Python 3.7 "
```

Comment:
Should we raise an exception here instead of warning? WDYT?

Reply:
`FutureWarning` is a step up from a `PendingDeprecationWarning` (and unlike a `PendingDeprecationWarning`, it can be seen by end users.)

Since for customers running on 3.7 or 3.8, next time they go to install, an Exception would be a breaking change for them ... I think it makes sense to stick with the `FutureWarning` and change to an `Exception` during the next major release.

Reply:
I can see about changing it to an Exception.